### PR TITLE
[Improvement] Get configuration values from environment data

### PR DESCRIPTION
**Pull request by mcop1**
## Changes in this pull request  
Resolves https://github.com/pimcore/copilot-bundle/issues/549

## Additional info
`extractConfigFieldFromJobStepConfig` is used to extract configuration values from the actual config. Adapted the method to check if the requested configuration is available in the environment data first. This allows to e.g. display a select-field before executing an action in Copilot, where the user can select from different endpoints to use. This can be useful for language translations since we need to use different endpoints for de->en, en->de, ... translations

### CODE REVIEW

**Comment by fashxp** [comment]
should we really force a match between config name and environment variable name? 
what if in an action there are two steps with same config name but I want to assign two different values to them ... e.g. two different model endpoints?

**Comment by mcop1** [comment]
Just for clarification: As internally discussed, we will use a certain method/placeholder in the configuration itself, that gets replaced while requesting the configuration value.

File: bundles/GenericExecutionEngineBundle/src/Messenger/Handler/AbstractAutomationActionHandler.php [hunk 1]
```diff
@@ -151,17 +151,47 @@ protected function getCurrentJobStepConfig(GenericExecutionEngineMessageInterfac
...
+
+        return $job === null ? [] : $job->getEnvironmentData();
+    }
+
+    protected function replaceConfigValueWithEnvVariable(
+        string $value,
+        array $variables
+    ): mixed {
+        /** @var $matches array */
+        if(!preg_match_all("/pimcore_copilot_get_env\('([^']*)'\)/", $value, $matches)) {
```
Comment (dpfaffenbauer) [inline bundles/GenericExecutionEngineBundle/src/Messenger/Handler/AbstractAutomationActionHandler.php]: I think it is weird that the Core refers to things from an extension.

Comment (mcop1) [reply bundles/GenericExecutionEngineBundle/src/Messenger/Handler/AbstractAutomationActionHandler.php]: For sure, you are absolutely right, the name should reflect the generic execution engine.

File: bundles/GenericExecutionEngineBundle/src/Messenger/Handler/AbstractAutomationActionHandler.php [hunk 2]
```diff
@@ -151,17 +151,47 @@ protected function getCurrentJobStepConfig(GenericExecutionEngineMessageInterfac
...
+
+        return $job === null ? [] : $job->getEnvironmentData();
+    }
+
+    protected function replaceConfigValueWithEnvVariable(
+        string $value,
+        array $variables
+    ): mixed {
+        /** @var $matches array */
+        if(!preg_match_all("/pimcore_gee_get_env\('([^']*)'\)/", $value, $matches)) {
```
Comment (fashxp) [inline bundles/GenericExecutionEngineBundle/src/Messenger/Handler/AbstractAutomationActionHandler.php]: why not just use `get_env`?

Comment (mcop1) [reply bundles/GenericExecutionEngineBundle/src/Messenger/Handler/AbstractAutomationActionHandler.php]: Didnt want anyone to confuse it with maybe existing methods and be as descriptive as possible. But I am open to suggestions ofc, so if you prefer the shorter version, then that´s fine with me too.

Comment (mcop1) [reply bundles/GenericExecutionEngineBundle/src/Messenger/Handler/AbstractAutomationActionHandler.php]: Internally discussed and settled for `job_env`

**Pull request merged**